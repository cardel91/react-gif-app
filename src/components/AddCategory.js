import React from 'react'
import { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {



    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const handleSubmit = (e)=>{
        e.preventDefault();

        // validacion de inputvalue
        if(inputValue.trim().length >2){
            // llamar con el callback del estado anterior
            setCategories(c=>[inputValue, ...c ]);
            setInputValue('');
        }
        // this.props.setCategories(e.target.value);
    }

    return (
        <form onSubmit={handleSubmit}>
            <input 
                type="text" 
                value = {inputValue}
                onChange = {handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
