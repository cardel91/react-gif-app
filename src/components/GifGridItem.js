import React from 'react'
import PropTypes from 'prop-types';

// var count = 0;
// export const GifGridItem = (props) => {
    // count++;
    // console.log(count);
export const GifGridItem = ({id, title, url}) => {
    // console.log(id, title, url);
    return (
        <div className='card animate__animated animate__bounce'>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    id: PropTypes.string,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}