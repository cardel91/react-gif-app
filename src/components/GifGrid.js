import React/*,{ useState, useEffect }*/ from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/GetGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    // const [images, setImages] = useState([]);


    
    // enviar category como prop
    // alias: images para propiedad data
    const {data: images, loading} = useFetchGifs(category);
    // console.log(state);
    return (

        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            { loading && <p className="animate__animated animate__flash">'Cargando...'</p>}
            <div className='card-grid'>
                {/* <ol> */}
                    {
                        images.map(img =>{
                        // desestructurar el objeto img
                        // images.map(({id, title, url}) =>{

                            return(
                                <GifGridItem key={img.id} 
                                // 
                                {...img}
                                />
                            );
                        })
                    }
                {/* </ol> */}
            </div>
        </>
    )
}
