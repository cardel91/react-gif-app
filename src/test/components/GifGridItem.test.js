import React from 'react'
import { GifGridItem }  from "../../components/GifGridItem";
import { shallow } from "enzyme";

describe('Pruebas en GifGridItem', () => {

    const title = "Titulo";
    const url = "http://localhost.jpg"

    const wrapper = shallow(<GifGridItem title={title} url={url} />);

   
    test('mostrar el componente', ()=> {
        expect(wrapper).toMatchSnapshot();
    });

    test('el titulo debe estar en un parrafo html', () => {
       
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    });

    test('debe tener imagen igual a url y alt de los props', () => {
        
        const img = wrapper.find('img');

        expect(img.props().src).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });

    test('debe tener animate__fadeIn', () => {
       
        const div = wrapper.find('div');
        // console.log(div.props());
        // expect(div).toHaveClas
        // expect(div).toHaveClass('animate__fadeIn');
        const className = div.prop('className');
        expect(className.includes('animate__bounce')).toBe(true);
    });
    
    
    

});
