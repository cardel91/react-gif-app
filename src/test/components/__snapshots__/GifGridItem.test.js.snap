// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem mostrar el componente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Titulo"
    src="http://localhost.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
